### PR TITLE
Add CandidateHandler and FilterHandler & tests; minor data model updates; update load_demo_data

### PR DESCRIPTION
**Pull request by acrellin**
Closes https://github.com/skyportal/skyportal/issues/408

### CODE REVIEW

**Comment by acrellin** [comment]
[ch749]

**Comment by acrellin** [comment]
[ch896]

**Comment by acrellin** [comment]
To keep this a manageable size, I'll open the front-end scanning page PR after this is merged.

File: skyportal/tests/conftest.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-'''Test fixture configuration.'''
+"""Test fixture configuration."""
```
Comment (acrellin) [inline skyportal/tests/conftest.py:1]: I ran `black` on this file to address some formatting issues

File: static/js/components/SearchBox.jsx [hunk 2]
```diff
@@ -242,9 +242,9 @@ const SearchBox = ({ sources }) => {
           <div style={{ display: "inline-block" }}>
             <i>
               Displaying&nbsp;
-              {sources.sourceNumberingStart}
+              {sources.numberingStart}
```
Comment (acrellin) [inline static/js/components/SearchBox.jsx:245]: These are renamed to make it easier to use the same query results pagination function for both the scanning page & source list GET requests

**Comment by stefanv** [comment]
@acrellin I thought I'd try out the branch before I reviewed the code, but I'm running into this problem, after doing `make db_init && make load_demo_data`:

```
sqlalchemy.exc.ProgrammingError: (psycopg2.errors.DuplicateObject) type "filters" already exists
HINT:  A relation has an associated type of the same name, so you must use a name that doesn't conflict with any existing type.
 [SQL: '\nCREATE TABLE filters (\n\tid SERIAL NOT NULL, \n\tcreated_at TIMESTAMP WITHOUT TIME ZONE NOT NULL, \n\tquery_string VARCHAR NOT NULL, \n\tgroup_id INTEGER, \n\tPRIMARY KEY (id), \n\tFOREIGN KEY(group_id) REFERENCES groups (id)\n)\n\n'] (Background on this error at: http://sqlalche.me/e/f405)
make: *** [Makefile:51: load_demo_data] Error 1
```

Do you know what may cause this?

**Comment by acrellin** [comment]
@stefanv no, I have no idea... I can't reproduce locally. I'll see if I can
gain any insight into what might be happening.

File: skyportal/handlers/api/candidate.py [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+    @auth_or_token
+    def get(self, obj_id=None):
+        """
+        ---
+        single:
+          description: Retrieve a candidate
+          parameters:
+            - in: path
+              name: obj_id
+              required: false
```
Comment (stefanv) [inline skyportal/handlers/api/candidate.py:30]: Is this not a required parameter for the single case?

File: skyportal/handlers/api/candidate.py [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+                application/json:
+                  schema: Error
+        """
+        if obj_id is not None:
+            c = Candidate.get_if_owned_by(obj_id, self.current_user)
+            if c is None:
+                return self.error("Invalid ID")
+            return self.success(data={"candidates": c})
+
+        page_number = self.get_query_argument("pageNumber", None) or 1
```
Comment (stefanv) [inline skyportal/handlers/api/candidate.py:60]: Where are these query arguments documented?

File: skyportal/handlers/api/filter.py [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+    @auth_or_token
+    def get(self, filter_id=None):
+        """
+        ---
+        single:
+          description: Retrieve a filter
+          parameters:
+            - in: path
+              name: filter_id
+              required: false
```
Comment (stefanv) [inline skyportal/handlers/api/filter.py:21]: Same question as for the other one: required?

**Pull request merged**